CI/include/build_support_windows.ps1

```diff
@@ -0,0 +1,579 @@
+            git submodule foreach --recursive git submodule sync
+            git submodule update --init --recursive
+        }
+    }
+}
+
+function GitHub-Fetch {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [String] $GH_USER,
+        [Parameter(Mandatory=$true)]
+        [String] $GH_REPO,
+        [Parameter(Mandatory=$true)]
+        [String] $GH_REF
+    )
+    if ($PSBoundParameters.Count -ne 3) {
+        Write-Error "Usage: GitHub-Fetch GITHUB_USER GITHUB_REPOSITORY GITHUB_REF"
+        return 1
+    }
+
+    Git-Fetch "https://github.com" "${GH_USER}" "${GH_REPO}" "${GH_REF}"
+}
+
+function GitLab-Fetch {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [String] $GL_USER,
+        [Parameter(Mandatory=$true)]
+        [String] $GL_REPO,
+        [Parameter(Mandatory=$true)]
+        [String] $GL_REF
+    )
+    if ($PSBoundParameters.Count -ne 3) {
+        Write-Error "Usage: GitLab-Fetch GITLAB_USER GITLAB_REPOSITORY GITLAB_REF"
+        return 1
+    }
+
+    Git-Fetch "https://gitlab.com" "${GL_USER}" "${GL_REPO}" "${GL_REF}"
+}
+
+function Apply-Patch {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [String] $COMMIT_URL,
+        [Parameter(Mandatory=$true)]
+        [String] $COMMIT_HASH
+    )
+
+    $PATCH_FILE = Get-Basename "${COMMIT_URL}"
+
+    if ("${COMMIT_URL}".Substring(0, 5) -eq "https") {
+        Invoke-WebRequest "${COMMIT_URL}" -UseBasicParsing -OutFile "${PATCH_FILE}"
+        if ("${COMMIT_HASH}" -eq $(Get-FileHash ${PATCH_FILE}).Hash) {
+            Write-Info "${PATCH_FILE} downloaded successfully and passed hash check"
+        } else {
+            Write-Error "${PATCH_FILE} downloaded successfully and failed hash check"
+            return 1
+        }
+    } else {
+        $PATCH_FILE = "${COMMIT_URL}"
+    }
+
+    Write-Info "Applying patch ${PATCH_FILE}"
+
+    if (Test-Path "./.git") {
+        git apply "${PATCH_FILE}"
+    } else {
+        # TODO: patch failed in my tests, hence using git apply above
+        patch -g 0 -f -p1 -i "${PATCH_FILE}"
+    }
+}
+
+function Check-Archs {
+    Write-Step "Check Architecture..."
+
+    if ("${BuildArch}" -eq "64-bit") {
+        $script:ARCH = "x86_64"
+        $script:CMAKE_ARCH = "x64"
+        $script:CMAKE_BITNESS = "64"
+        $script:CMAKE_INSTALL_DIR = "win64"
+    } elseif ("${BuildArch}" -eq "32-bit") {
+        $script:ARCH = "x86"
+        $script:CMAKE_ARCH = "Win32"
+        $script:CMAKE_BITNESS = "32"
+        $script:CMAKE_INSTALL_DIR = "win32"
+    } else {
+        Caught-Error "Unsupported architecture '${BuildArch}' provided"
+    }
+}
+
+function Check-Curl {
+    if (!(Test-CommandExists "curl.exe") -and !$NoChoco) {
+        Write-Step "Install curl from chocolatey..."
+        Invoke-Expression "choco install -y curl"
+    }
+
+    # TODO: implement a way to force using/installing curl.exe from chocolatey
+    #$CURLCMD = "C:\ProgramData\chocolatey\bin\curl.exe"
```

Comment:
Hmm.  I'd actually left this in as a reminder of where Chocolatey installs curl, but I see that because it's originally a line of code its purpose is confusing.  I might leave the path in place, but try to make it more part of the TODO entry rather than a pure line of code simply commented out.